src/d/actor/d_a_tag_kb_item.cpp

```diff
@@ -1,58 +1,131 @@
+    int _create();
+    bool _execute();
+    bool _draw();
+
+    /* 0x290 */ u8 field_0x290[0x298 - 0x290];
+    /* 0x298 */ u8 field_0x298;
+    /* 0x299 */ u8 field_0x299;
+    /* 0x29C */ s32 field_0x29c;
+    /* 0x2A0 */ u8 field_0x2a0;
+    /* 0x2A1 */ u8 field_0x2a1;
+    /* 0x2A4 */ s32 field_0x2a4;
+    /* 0x2A8 */ fopAc_ac_c* mpActor;
+};  // Size: 0x2AC
 
 /* 00000078-000000C8       .text _delete__13daTagKbItem_cFv */
-void daTagKbItem_c::_delete() {
-    /* Nonmatching */
+bool daTagKbItem_c::_delete() {
+#if VERSION != VERSION_JP
+    if (field_0x2a0 != 0xff && field_0x2a4 != 0xff) {
+        dComIfGs_offSwitch(field_0x2a4, orig.roomNo);
+    }
+#endif
+    return 1;
 }
 
 /* 000000C8-0000010C       .text CreateInit__13daTagKbItem_cFv */
 void daTagKbItem_c::CreateInit() {
-    /* Nonmatching */
+    field_0x2a0 = fopAcM_GetParam(this);
+    s8 paramBit = fopAcM_GetParamBit(this, 8, 8);
+    field_0x29c = paramBit;
+    field_0x2a1 = fopAcM_GetParamBit(this, 0x10, 8);
+    field_0x2a4 = fopAcM_GetParamBit(this, 0x18, 8);
+    field_0x298 = 0;
+    field_0x299 = 0;
+    mpActor = NULL;
 }
 
 /* 0000010C-000001BC       .text _create__13daTagKbItem_cFv */
-void daTagKbItem_c::_create() {
-    /* Nonmatching */
+int daTagKbItem_c::_create() {
+    fopAcM_SetupActor(this, daTagKbItem_c);
+
+    CreateInit();
+    if (field_0x29c != 0x1f && dComIfGs_isItem(field_0x29c, orig.roomNo) ||
+        field_0x2a4 != 0xff && dComIfGs_isSwitch(field_0x2a4, orig.roomNo))
+    {
+        return cPhs_ERROR_e;
+    }
+    return cPhs_COMPLEATE_e;
 }
 
 /* 000001BC-000001C4       .text _execute__13daTagKbItem_cFv */
-void daTagKbItem_c::_execute() {
-    /* Nonmatching */
+bool daTagKbItem_c::_execute() {
+#if VERSION == VERSION_JP
+    if (field_0x29c != 0x1f && dComIfGs_isItem(field_0x29c, orig.roomNo) ||
+        field_0x2a4 != 0xff && dComIfGs_isSwitch(field_0x2a4, orig.roomNo))
+    {
+        fopAcM_delete(this);
+    }
+#endif
+    return 1;
 }
 
 /* 000001C4-000001CC       .text _draw__13daTagKbItem_cFv */
-void daTagKbItem_c::_draw() {
-    /* Nonmatching */
+bool daTagKbItem_c::_draw() {
+    return 1;
 }
 
 /* 000001CC-000001EC       .text daTagKbItem_Create__FPv */
-void daTagKbItem_Create(void*) {
-    /* Nonmatching */
+static int daTagKbItem_Create(void* i_this) {
+    return static_cast<daTagKbItem_c*>(i_this)->_create();
 }
 
 /* 000001EC-00000210       .text daTagKbItem_Delete__FPv */
-void daTagKbItem_Delete(void*) {
-    /* Nonmatching */
+static int daTagKbItem_Delete(void* i_this) {
+    return static_cast<daTagKbItem_c*>(i_this)->_delete();
 }
 
 /* 00000210-00000234       .text daTagKbItem_Draw__FPv */
-void daTagKbItem_Draw(void*) {
-    /* Nonmatching */
+static int daTagKbItem_Draw(void* i_this) {
+    return static_cast<daTagKbItem_c*>(i_this)->_draw();
 }
 
 /* 00000234-00000258       .text daTagKbItem_Execute__FPv */
-void daTagKbItem_Execute(void*) {
-    /* Nonmatching */
+static int daTagKbItem_Execute(void* i_this) {
```

Comment:
ideally these return types should match the return type of the member function

Reply:
not in this case though. Objdiff shows clrlwi's that need to be generated in the static execute and static draw function. It doesn't match else

Reply:
hm, ill approve for now but i want to take a closer look at that later